frontend/src/static/js/components/webstatus-sidebar-menu.ts

```diff
@@ -129,25 +132,21 @@ export class WebstatusSidebarMenu extends LitElement {
   }
 
   @state()
-  bookmarks: Bookmark[] = [];
+  private activeBookmarkQuery: string | null = null;
 
+  @consume({context: appBookmarkInfoContext, subscribe: true})
   @state()
-  private activeBookmarkQuery: string | null = null;
+  appBookmarkInfo?: AppBookmarkInfo;
 
   // For now, unconditionally open the features dropdown.
   @state()
   private isFeaturesDropdownExpanded: boolean = true;
 
   updateActiveStatus(): void {
```

Comment:
`updateActiveStatus` here is updated when there is a `popstate` event, while `this._currentGlobalBookmark` is also updated when `private handlePopState() {` is called in `webstatus-bookmarks-service.ts`. Would there be a sync issue? How do we make sure that `this.appBookmarkInfo?.currentGlobalBookmark?.query` is already updated when this function is called

Reply:
This is a good call. Instead, I should remove handlePopState and instead use willUpdate on the context.

Also, I might bring in some of my helpers from this [branch](https://github.com/GoogleChrome/webstatus.dev/compare/jcscottiii/ui-searches-v3...jcscottiii/ui-saved-searches-v4?expand=1) too.

Reply:
This is resolved